### PR TITLE
Movie Details page: Review List

### PR DESCRIPTION
**Pull request by saeraa**
Really not sure about the tests, but .. at least there are tests, right. 🤷🏻‍♀️

### CODE REVIEW

File: static/js/reviewList.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	if (currentPage == totalPages) {
+		buttonNextEl.disabled = true;
+	} else {
+		buttonNextEl.disabled = false;
+	}
+}
+
+async function onPageLoad() {
+	const reviews = await getReviewsFromAPI(movieID);
+	console.log(reviews);
```
Comment (Arash-SJPI22) [inline static/js/reviewList.js]: I think you forgot to remove a bug hunting console.log :)

File: static/js/reviewList.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+	date.textContent = dateCreated.toLocaleDateString("sv-SE");
+
+	reviewEl.append(rating, author, date, comment);
+
+	return reviewEl;
+}
+
+async function getReviewsFromAPI(movieID, page = currentPage) {
+	const response = await fetch(
+		`http://localhost:5080/api/movies/${movieID}/reviews?page=${page}`
```
Comment (Arash-SJPI22) [inline static/js/reviewList.js]: Would be better to change it to `/api/movies/${movieID}/reviews?page=${page}` insted so if we deploy, it would still work without the need to edit this line.

**Review by Arash-SJPI22 — CHANGES_REQUESTED** [review]
Very nice review-hats :dancers: 

/static/js/reviewList.js is a bit messy with many functions and jumps, it's was a bit hard to follow to the code. But it works and only if you find joy in it could you modulate it :D (it's not a merge blocker)

I think you should move "movies.js" from "/static/js/" to "/src/" , since users of the page should not be able to get it because it contains a backend API call.
**I have done this for my branch that I am working on and think you should do it to before PR is merged.**

File: static/js/reviewList.js (hunk 1 above)
Comment (saeraa) [reply static/js/reviewList.js]: Nice catch! 🪲

File: static/js/reviewList.js (hunk 2 above)
Comment (saeraa) [reply static/js/reviewList.js]: Yes! Excellent suggestion.

**Comment by saeraa** [comment]
> 



I've tried to separate the logic out to different functions so it would be easier to read. I tend to look at the code like this, with block collapsed for easy overview: 


I'm not sure I'm following what you mean by it being messy. I'd be interested to know how you think the code could be improved upon!

I've made changes for the other suggestions you made!

**Review by Arash-SJPI22 — APPROVED** [review]
Great work! :dango:

**Comment by Arash-SJPI22** [comment]
> > 
> 
> I've tried to separate the logic out to different functions so it would be easier to read. I tend to look at the code like this, with block collapsed for easy overview
> 
> I'm not sure I'm following what you mean by it being messy. I'd be interested to know how you think the code could be improved upon!
> 
> I've made changes for the other suggestions you made!

I haven't really used block collapse when coding before, it helped!

After trying to rewrite the code and then checking my own code, witch has the functions in different files and after the weekend I forgot how my code worked and jumping from file to file to follow the flow wasn't much clearer... So I humbly withdraw my comment :bow:

File: src/reviewList.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+};
+
+const getReviewsForMovieWithId = async (
+	movieId,
+	pageNumber,
+	callApiFunction = apiAdapter
+) => {
+	try {
+		const response = await callApiFunction(movieId, pageNumber);
+		// if, for some reason, the pagination isn't working from the API, slice the review results to 5 ??? probably pointless, who knows
```
Comment (ved0) [inline src/reviewList.js:24]: maybe have something like const payload = response.data. So it is not data.data later in the code :smile:

File: scss/reviewList.scss [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+		margin: 0.8rem 0;
+		background-color: rgba($c-red-light, 0.6);
+		background-color: $c-gray-light;
+		box-shadow: 0px 6px 12px 0px rgb(0 0 0 / 20%);
+
+		.review-rating {
+			display: flex;
+			padding-bottom: 0.2rem;
+			.review-hat.filled {
+				background-image: url("/img/hat-fill.png");
```
Comment (ved0) [inline scss/reviewList.scss:48]: nice hats! 🎅

File: static/js/reviewList.js [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+		currentPage--;
+	}
+	const reviews = await getReviewsFromAPI(movieID);
+	insertReviewsInHTML(reviews);
+	disableButtons();
+}
+
+function disableButtons() {
+	if (currentPage == 1) {
+		buttonPrevEl.disabled = true;
```
Comment (ved0) [inline static/js/reviewList.js:32]: useful attribute! 💯

File: src/api.js [hunk 6]
```diff
@@ -1,9 +1,17 @@
...
+import { getReviewsForMovieWithId } from "./reviewList.js";
 
 const apiRouter = express.Router();
 
 apiRouter.get("/movies/:id", (req, res) => {
 	res.send(`movie with id of ${req.params.id}`);
 });
 
+apiRouter.get("/movies/:id/reviews", async (req, res) => {
+	const page = req.query.page;
```
Comment (richardolsson) [inline src/api.js:11]: Could also be written `const { page } = req.query`, a syntax which is preferred by many.

File: src/reviewList.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+		const reviewList =
+			response.data.data.length > 5
+				? response.data.data.slice(0, 5)
+				: response.data.data;
```
Comment (richardolsson) [inline src/reviewList.js:28]: This could actually be simplified as just `const reviewList = response.data.data.slice(0, 5);`.

The `slice()` method will gracefully just handle the case where length is shorter than 5, and return the full array in that case, even if it's shorter than 5.

**Review by richardolsson — COMMENTED** [review]
I think this looks great! I personally don't like the tab indentation because I think it gets too wide in the github interface, but that's just a personal preference. I have very little critique, but left two comments below.
